### PR TITLE
Bumps version to v1.0.0-beta4

### PR DESCRIPTION
**Pull request by mohammedzee1000**
This PR bumps the odo version to v1.0.0-beta4

# Release Notes Draft

## Breaking Changes
- The openshift labels used internally by odo to identify its components, have been updated to match https://github.com/gorkem/app-labels/blob/master/labels-annotation-for-openshift.adoc. This breaks backward compatibility with older versions of odo as components created by older versions will no longer be identified https://github.com/openshift/odo/pull/1959

## Enhancements
 - Removes an extra `exist` call and removes the spinner around it https://github.com/openshift/odo/pull/1965
- Graceful recovery from panic with mesage https://github.com/openshift/odo/pull/1954
- Fixed failing push when numeric directory is given https://github.com/openshift/odo/pull/1933
- Remove error if app does not have component https://github.com/openshift/odo/pull/1919
- `odo config set` no longer creates local config, instead reporting error https://github.com/openshift/odo/pull/1900
- No logs on`--output` flag as it is now global https://github.com/openshift/odo/pull/1898

## Testing
- Added gosec static code analysis https://github.com/openshift/odo/pull/1958
- Fixed `oc describe dc` test flakes https://github.com/openshift/odo/pull/1963
- Enabled component sub test in Makefile https://github.com/openshift/odo/pull/1962
- Add message with program and arguments to helper_run.go output https://github.com/openshift/odo/pull/1948
- Fixed flaky unit tests due to reordering of port lists https://github.com/openshift/odo/pull/1924
- Fixed false positives in `CmdShouldFail` https://github.com/openshift/odo/pull/1902
- All tests except login test now run in parallel

## Internal
- `SourceLocation` of binary when `--context` flag is used is now saved relative to component context https://github.com/openshift/odo/pull/1921
- Optimised `component.Exists` https://github.com/openshift/odo/pull/1905
- Component spec now includes project name https://github.com/openshift/odo/pull/1904 

## Documentation
- Fixed the help message for `--path ` flag in `odo list ` https://github.com/openshift/odo/pull/1973
- Added newline spacing to `odo app --help` https://github.com/openshift/odo/pull/1966
- Updated wildfly source location to https://github.com/wildfly/wildfly-s2i https://github.com/openshift/odo/pull/1906
- Added documentation on the `GLOBALODOCONFIG` environment variable https://github.com/openshift/odo/pull/1939
- Adding minor nodes for new developers https://github.com/openshift/odo/pull/1926
- Fixed backticks and grammatical errors https://github.com/openshift/odo/pull/1901
- Fixed broken links in getting started https://github.com/openshift/odo/pull/1896

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

File: docs/cli-reference.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-= Overview of the odo (OpenShift Do) CLI Structure
+= Overview of the Odo (OpenShift Do) CLI Structure
```
Comment (kadel) [inline docs/cli-reference.adoc]: The typography standard that we agreed on is that odo will be written with lowercase `o` :-)

Comment (kadel) [reply docs/cli-reference.adoc]: Oh, this comes from the template, right? Then it should be updated.

File: docs/cli-reference.adoc [hunk 2]
```diff
@@ -625,22 +627,14 @@ _________________
   odo preference unset  Timeout
 ----
 
-Modifies odo specific configuration settings within the global preference file.
-
-[NOTE]
-====
-By default the path to the global preference file is `~/.odo/preferece.yaml`, it is stored in in the environment variable `GLOBALODOCONFIG`. You can set up a custom path by setting the value of the environment variable to a new preference path, for example `GLOBALODOCONFIG="new_path/preference.yaml"`
```
Comment (kadel) [inline docs/cli-reference.adoc]: Why was this removed?

File: docs/cli-reference.adoc (hunk 1 above)
Comment (mohammedzee1000) [reply docs/cli-reference.adoc]: yes

File: docs/cli-reference.adoc (hunk 2 above)
Comment (mohammedzee1000) [reply docs/cli-reference.adoc]: This was done by the script, I can add it back, but it also means script needs to be updated

Comment (mohammedzee1000) [reply docs/cli-reference.adoc]: Added back, but er should probably fix the scripts/template accordingly

File: docs/cli-reference.adoc (hunk 1 above)
Comment (mohammedzee1000) [reply docs/cli-reference.adoc]: Added back

**Comment by kadel** [comment]
> ## Breaking Changes
> * The openshift labels used internally by odo to identify its components, have been updated to match https://github.com/gorkem/app-labels/blob/master/labels-annotation-for-openshift.adoc. This breaks backward compatibility with older versions of odo as components created by older versions will no longer be identified #1959

I think that there is one more breaking change regarding ports -  https://github.com/openshift/odo/issues/1910

**Comment by kadel** [comment]
/approve
is this still WIP?

File: docs/cli-reference.adoc [hunk 3]
```diff
@@ -10,12 +10,12 @@ ___________________
   git clone https://github.com/openshift/nodejs-ex && cd nodejs-ex
   odo create nodejs
   odo push
-
+
```
Comment (amitkrout) [inline docs/cli-reference.adoc]: Unnecessary extra space. Remove the same in other places too

Comment (mohammedzee1000) [reply docs/cli-reference.adoc]: This is generated

**Comment by mohammedzee1000** [comment]
Windows side looking good, although a little hard to setup without cygwin
